### PR TITLE
Construct explanatory string of NotValidErrorType exception in ctor

### PR DESCRIPTION
**Pull request by hikinggrass**
This fixes clangs warning: returning address of local temporary object [-Wreturn-stack-address] compiler warning

### CODE REVIEW

File: include/utils/error.hpp [hunk 1]
```diff
@@ -24,13 +24,15 @@ RequestClearErrorOption string_to_request_clear_error_option(const std::string&
 
 class NotValidErrorType : public std::exception {
 public:
-    explicit NotValidErrorType(const std::string& error_type_) : error_type(error_type_){};
-    virtual const char* what() const throw() {
-        return fmt::format("Error type '{}' is not valid.", error_type).c_str();
+    explicit NotValidErrorType(const std::string& error_type_) :
+        error_type(error_type_), explanatory_string(fmt::format("Error type '{}' is not valid.", error_type)){};
```
Comment (a-w50) [inline include/utils/error.hpp]: Is there any reason, why `error_type` should be kept as a member variable?

**Review by a-w50 — APPROVED** [review]

File: include/utils/error.hpp (hunk 1 above)
Comment (hikinggrass) [reply include/utils/error.hpp]: @andistorm ?

Comment (andistorm) [reply include/utils/error.hpp]: `error_type` as member variable might allows to handle the exception in specific cases. Right now there is no usage of that. And it should be probably public to allow using it..

File: include/utils/error.hpp [hunk 2]
```diff
@@ -24,13 +24,15 @@ RequestClearErrorOption string_to_request_clear_error_option(const std::string&
...
+    const char* what() const noexcept override {
+        return explanatory_string.c_str();
     }
 
 private:
     const std::string error_type;
```
Comment (andistorm) [inline include/utils/error.hpp]:
```suggestion
    const std::string error_type;

private:
```

**Review by andistorm — APPROVED** [review]

File: include/utils/error.hpp (hunk 2 above)
Comment (hikinggrass) [reply include/utils/error.hpp]: I'm not so sure about making this member public, should probably be a getter. Or we remove it for now if there's no usage

Comment (a-w50) [reply include/utils/error.hpp]: I would remove it here, if it will be needed, it can just be added

**Pull request merged**